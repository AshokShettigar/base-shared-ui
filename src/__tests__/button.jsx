// import { Simulate, renderIntoDocument } from 'react-addons-test-utils';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { Button, BUTTON_SIZE, BUTTON_VARIATION } from 'platform-common-ui';

const defaultProps = {
    isDark: false,
    isDisabled: false,
    size: BUTTON_SIZE.MEDIUM,
    type: 'button',
    variation: BUTTON_VARIATION.DEFAULT,
    isLoading: undefined,
    onClick: jest.fn()
};

describe('button', function() {
    it('should render the button', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(
            <Button
                onClick={onClickHandler}
            >
                MyButton
            </Button>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a loading button', () => {
        const wrapper = shallow(<Button {...defaultProps} isLoading />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the button with any element', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(
            <Button
                onClick={onClickHandler}
            >
                <div>My Custom Label</div>
            </Button>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render with a custom class name', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(
            <Button
                className="my-class"
                onClick={onClickHandler}
            >
                MyButton
            </Button>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a disabled button', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(
            <Button
                onClick={onClickHandler}
                isDisabled
            >
                MyButton
            </Button>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a primary button', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(
            <Button
                onClick={onClickHandler}
                variation={BUTTON_VARIATION.PRIMARY}
            >
                MyButton
            </Button>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a secondary button', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(
            <Button
                onClick={onClickHandler}
                variation={BUTTON_VARIATION.SECONDARY}
            >
                MyButton
            </Button>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a small size button', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(
            <Button
                onClick={onClickHandler}
                size={BUTTON_SIZE.SMALL}
            >
                MyButton
            </Button>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a medium size button', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(
            <Button
                onClick={onClickHandler}
                size={BUTTON_SIZE.MEDIUM}
            >
                MyButton
            </Button>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a large size button', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(
            <Button
                onClick={onClickHandler}
                size={BUTTON_SIZE.LARGE}
            >
                MyButton
            </Button>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a lighter button for dark background', function() {
        const onClickHandler = () => true;
        const tree = renderer.create(
            <Button
                onClick={onClickHandler}
                isDark
            >
                MyButton
            </Button>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // it('should trigger onClick handler', function(done) {
    //     const onClickHandler = () => done();
    //     const component = renderIntoDocument(<Button onClick={onClickHandler}>MyButton</Button>);
    //     Simulate.click(component.refs.button);
    // });
});
