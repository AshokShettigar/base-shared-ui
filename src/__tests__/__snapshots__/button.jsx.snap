// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should render a disabled button 1`] = `
<button
  className="cn-btn cn-btn--disabled cn-btn--medium"
  data-component="Button"
  disabled={true}
  form={undefined}
  onClick={[Function]}
  type="button"
>
  MyButton
</button>
`;

exports[`button should render a large size button 1`] = `
<button
  className="cn-btn cn-btn--large"
  data-component="Button"
  disabled={false}
  form={undefined}
  onClick={[Function]}
  type="button"
>
  MyButton
</button>
`;

exports[`button should render a lighter button for dark background 1`] = `
<button
  className="cn-btn cn-btn--dark cn-btn--medium"
  data-component="Button"
  disabled={false}
  form={undefined}
  onClick={[Function]}
  type="button"
>
  MyButton
</button>
`;

exports[`button should render a loading button 1`] = `
<button
  className="cn-btn cn-btn--disabled cn-btn--medium"
  data-component="Button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <Spinner
    className="cn-modal-controls__button-spinner"
    size="extra-small"
  />
</button>
`;

exports[`button should render a medium size button 1`] = `
<button
  className="cn-btn cn-btn--medium"
  data-component="Button"
  disabled={false}
  form={undefined}
  onClick={[Function]}
  type="button"
>
  MyButton
</button>
`;

exports[`button should render a primary button 1`] = `
<button
  className="cn-btn cn-btn--medium cn-btn--primary"
  data-component="Button"
  disabled={false}
  form={undefined}
  onClick={[Function]}
  type="button"
>
  MyButton
</button>
`;

exports[`button should render a secondary button 1`] = `
<button
  className="cn-btn cn-btn--medium cn-btn--secondary"
  data-component="Button"
  disabled={false}
  form={undefined}
  onClick={[Function]}
  type="button"
>
  MyButton
</button>
`;

exports[`button should render a small size button 1`] = `
<button
  className="cn-btn cn-btn--small"
  data-component="Button"
  disabled={false}
  form={undefined}
  onClick={[Function]}
  type="button"
>
  MyButton
</button>
`;

exports[`button should render the button 1`] = `
<button
  className="cn-btn cn-btn--medium"
  data-component="Button"
  disabled={false}
  form={undefined}
  onClick={[Function]}
  type="button"
>
  MyButton
</button>
`;

exports[`button should render the button with any element 1`] = `
<button
  className="cn-btn cn-btn--medium"
  data-component="Button"
  disabled={false}
  form={undefined}
  onClick={[Function]}
  type="button"
>
  <div>
    My Custom Label
  </div>
</button>
`;

exports[`button should render with a custom class name 1`] = `
<button
  className="cn-btn cn-btn--medium my-class"
  data-component="Button"
  disabled={false}
  form={undefined}
  onClick={[Function]}
  type="button"
>
  MyButton
</button>
`;
